lib/bundler/cli/change.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Bundler
+  class CLI::Change
+    def initialize(options, gem_name)
+      @gem_name = gem_name
+      @options = options
+    end
+
+    def run
+      raise InvalidOption, "Please supply atleast one option to change." unless @options[:group] || @options[:version] || @options[:source]
+
+      definition = Bundler.definition
+
+      @dep = definition.dependencies.find {|d| d.name == @gem_name }
+
+      raise InvalidOption, "`#{@gem_name}` could not be found in the Gemfile." unless @dep
+
+      @pass_options = {}
+
+      initial_gemfile = IO.readlines(Bundler.default_gemfile)
+
+      set_group_options
+
+      set_version_options
+
+      @pass_options["source"] = @options[:source] if @options[:source]
+
+      begin
+        require "bundler/cli/remove"
+        CLI::Remove.new([@gem_name], {}).run
+
+        require "bundler/cli/add"
+        CLI::Add.new(@pass_options, [@gem_name]).run
+      rescue StandardError => e
+        Bundler.ui.error e
```

Comment:
should the error be re-raised instead, after restoring the initial gemfile?

Reply:
It is shown as error do we need to raise an exception?

Reply:
yes, we should let the `FriendlyErrors` module handle showing the error to the user, and ensure we exit with the proper exit code